clients/docs/java.rst

```diff
@@ -0,0 +1,317 @@
+.. _client-examples-java:
+
+Java
+====
+
+In this tutorial, you will run a Java client application that produces
+messages to and consumes messages from an |ak-tm| cluster.
+
+.. include:: includes/client-example-overview.rst
+
+
+Prerequisites
+-------------
+
+Client
+~~~~~~
+
+-  Java 1.8 or higher to run the demo application.
+
+-  Maven to compile the demo application.
+
+
+Kafka Cluster
+~~~~~~~~~~~~~
+
+.. include:: includes/client-example-prerequisites.rst
+
+
+Setup
+-----
+
+#. Clone the `confluentinc/examples GitHub repository
+   <https://github.com/confluentinc/examples>`__ and check out the
+   :litwithvars:`|release|-post` branch.
+
+   .. codewithvars:: bash
+
+      git clone https://github.com/confluentinc/examples
+      cd examples
+      git checkout |release|-post
+
+#. Change directory to the example for Java.
+
+   .. code-block:: bash
+
+      cd clients/cloud/java/
+
+#. .. include:: includes/client-example-create-file-java.rst
+
+
+Basic Producer and Consumer
+---------------------------
+
+In this example, the producer writes Kafka data to a topic in your Kafka
+cluster. Each record has a key representing a username (for example, ``alice``)
+and a value of a count, in JSON format (for example, ``{"Count": 0}``). The
+consumer reads the same topic and keeps a rolling sum of the counts as it
+processes each record.
+
+
+Produce Records
+~~~~~~~~~~~~~~~
+
+#. Run the producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
```

Comment:
Let's make Compile the java code and `mvn clean package` its own numbered bullet.

So, above, before, `Run the producer...` add

```
#. Compile the java code

    .. code-block:: bash

      mvn clean package
```

Then you can remove all instances of this here, and below.